tests/cases/standalone/common/promql/scalar.sql

```diff
@@ -101,4 +101,46 @@ TQL EVAL (0, 15, '5s') scalar(scalar(host{host="host2"}) + 1);
 
 TQL EVAL (0, 15, '5s') scalar(scalar(host{host="host1"}) + scalar(host{host="host2"}));
 
+-- Test clamp functions with vector input and scalar bounds
+
+-- SQLNESS SORT_RESULT 3 1
+TQL EVAL (0, 15, '5s') clamp(host, 0, 12);
+
+-- SQLNESS SORT_RESULT 3 1
+TQL EVAL (0, 15, '5s') clamp(host{host="host1"}, -1, 6);
+
+-- SQLNESS SORT_RESULT 3 1
+TQL EVAL (0, 15, '5s') clamp_min(host{host="host1"}, 10);
+
+-- SQLNESS SORT_RESULT 3 1
+TQL EVAL (0, 15, '5s') clamp_min(host{host="host1"}, 1);
+
+-- SQLNESS SORT_RESULT 3 1
+TQL EVAL (0, 15, '5s') clamp_max(host{host="host1"}, 1);
+
+-- SQLNESS SORT_RESULT 3 1
+TQL EVAL (0, 15, '5s') clamp_max(host{host="host1"}, 10);
+
+-- SQLNESS SORT_RESULT 3 1
+TQL EVAL (0, 15, '5s') clamp_min(host, 1);
+
+-- SQLNESS SORT_RESULT 3 1
+TQL EVAL (0, 15, '5s') clamp_max(host, 10);
```

Comment:
We can refine those test cases to make them minimal and focused. E.g., only one or two cases for filters. `TQL EVAL (0, 15, '5s') clamp_max(host, 10);` and `TQL EVAL (0, 15, '5s') clamp_max(host{host="host1"}, 10);` are overlapping

Reply:
It's fine. Although they overlap, one is for filtering and the other is not.